src/buf/into_buf.rs

```diff
@@ -63,6 +67,7 @@ impl<'a> IntoBuf for &'a [u8] {
     }
 }
 
+#[cfg(feature = "std")]
 impl<'a> IntoBuf for &'a mut [u8] {
     type Buf = io::Cursor<&'a mut [u8]>;
```

Comment:
To really support `alloc` properly I think these `impl`s (both this one and the one below for `&'a str` need to change to not use an `io::Cursor`), however as `Buf` is an associated type, I imagine it's considered part of the public API and changes to the underlying `Buf` type would be considered breaking.

As a breaking change I think ideally we could fix this by eliminating all usages of `io::Cursor` from the crate (which, I believe, was @carllerche's plan for v0.5.x anyway). However I'd like to stress that even without these impls, bytes is still useful, because downstream `no_std` + `alloc` consumers of `bytes` can define their own buffer type (possibly just vendoring `io::Cursor` from `libstd` and using the existing `std`-dependent impls) and then `imp FromBuf` and `impl IntoBuf` for their own type. This would be my planned approach for now if this PR were to be merged.